src/harness/test_harness_objects.py

```diff
@@ -0,0 +1,327 @@
+    heartbeat_requests = []
+    for grant_object in self.grant_objects.itervalues():
+      if grant_object.isActive():
+        heartbeat_requests.append(grant_object.constructHeartbeatRequest())
+    return heartbeat_requests
+
+  def getOperationParamsOfAllAuthorizedGrants(self):
+    """Returns the list of operation params of all authorized grants."""
+    operation_params = []
+    for grant_object in self.grant_objects.itervalues():
+      if grant_object.isGrantAuthorizedInLastHeartbeat():
+        operation_params.append(grant_object.getRequestOperationParam())
+    return operation_params
+
+
+class DomainProxy(object):
+  """Performs Domain Proxy related operations.
+
+  This class stores Domain Proxy related information like the SSL certificate, key to be used and the set of
+  CBSD objects belonging to this Domain Proxy. Bulk operations like registration, grant and heartbeat
+  procedures are performed on all of the CBSDs belonging to this Domain Proxy.
+  """
+  # Define an enumeration class named ResponseCodes with members
+  # 'SUCCESS'
+  class RelinquishmentResponseCodes(Enum):
+    SUCCESS = 0
+
+  def __init__(self, ssl_cert, ssl_key, testcase):
+    """
+    Args:
+      ssl_cert: Path to SSL cert file.
+      ssl_key: Path to SSL key file.
+      testcase: test case object from the caller.
+    """
+    self.ssl_cert = ssl_cert
+    self.ssl_key = ssl_key
+    self.cbsd_objects = {}
+    self.testcase = testcase
+
+  def addCbsdsAndGrants(self, registration_requests, grant_requests):
+    """Construct CBSD object based on the result of registration and
+    grant requests.
+
+    Args
+      registration_requests: A list of dictionary elements ,where each list elements is
+        individual CBSD registration requests (each of which is itself a dictionary).
+      grant_requests: A list of dictionary elements,where each list elements is
+        individual CBSD grant requests (each of which is itself a dictionary).
+    """
+    # Checking if the number of registration requests matches number of grant requests.
+    # There should be exactly one grant request per registration request.
+    assert len(grant_requests) == len(registration_requests)
+    cbsd_ids = self.testcase.assertRegistered(registration_requests,
+                                              cert=self.ssl_cert, key=self.ssl_key)
+
+    # Copy the cbsdId from Registration response to grant requests.
+    for index, grant_request in enumerate(grant_requests):
+      grant_request['cbsdId'] = cbsd_ids[index]
+
+    # Wrapped version of the grant requests.
+    grant_requests_wrap = {
+      'grantRequest': grant_requests
+    }
+
+    #perform grant operation.
+    grant_responses = self.testcase._sas.Grant(grant_requests_wrap,
+                                               self.ssl_cert, self.ssl_key)['grantResponse']
+
+    # Make one CBSD object per successful grant request.
+    for index, grant_request in enumerate(grant_requests):
```

Comment:
zip takes an arbitrary number of arguments. Using it will make the code in this loop more readable.
```python
zip(grant_requests, grant_responses, registration_requests, cbsd_ids)
```

Reply:
Agreed we can use zip, thanks for suggestion.